### DISCUSSION TITLE
API usage as a buyer (+ please approve my app, requested > 1 month ago!)

### DISCUSSION THREAD

**Discussion by laoraryn** [etsy/open-api#1278] [Q&A] [answered]
Hi, all. I don't have an Etsy shop, I just buy things. I'm trying to write a Python program that periodically updates me on prices and availability of things on my wishlist

Etsy flagged my program as a bot, so I said, ok, I'll check out the API to do it the "right" way. As far as I can see, the API isn't meant for buyers, only sellers. However it looks like I could still use the getListing endpoint for what I need

Is this going to help me not get blocked by the site's anti-spam guards? If not, what's the preferred way for me to do this? It would be 100-200 public page views once or twice a week, max, it's not like I'm going to be bogging anything down.

On a related note, I put in a request over a month ago, and it's still showing as pending approval. How long does this process usually take??

**Comment by turboturtleD** [comment, marked as answer]
So, screen scraping is explicitly against the terms of service, so that’s maybe what caused your previous issue (not sure how else you would access data without the API).  Etsy wants to control what data is allowed to be accessed programmatically, and this does hamstring a lot of developers’ projects.

While I think the Etsy API is largely “tailored” for sellers, I think there is reasonable API access to create some level of app success for buyer(s) also.  The biggest caveat is that Etsy has a VERY narrow vision of what should and shouldn’t be allowed in regards to app building.  They don’t want anyone to offer an alternative “Etsy experience” and seem inconsistent and unclear (at best) in informing developers what is and isn’t acceptable for a developer app.

In regards to rate limiting, there should be no concerns about “public views” beyond just that (rate limiting).  There is API documentation for rate limiting, I don’t know the numbers off the top of my head though.

As far as the rejection process (oh… I mean approval process), I think a few weeks time is pretty normal, a month or so would not be shocking.  I would recommend digging into the GitHub discussions if your app is not ultimately accepted for some ideas for what is required to get your app approved.  Unfortunately, there is not a better resource.

**Reply by laoraryn** [reply]
Thanks for a quick reply! To be clear, I'm using "app" since that's the word they use in documentation, but it's just me hacking together a .py file on my laptop, I can guarantee no one but me will ever use it. I marked it as "for personal use" in the application process, which seems to have been the problem for a lot of people online.

Does anyone from Etsy ever get on these issues and offer guidance, or is it all community stuff?

**Reply by turboturtleD** [reply]
On rare occasion, someone from Etsy pops out of the woodwork, however not too often these days.

The word “app” really refers simply to whatever you are using to access the API, even if that’s just Postman or some personal homebrew Python script.

It’s definitely by far easier to get personal access, but that doesn’t mean it’s an automatic “yes”.

It wouldn’t hurt to send the developers email a message and ask about the status of your submission.  Pretty sure it’s developers@etsy.com (or developer@etsy.com).

**Reply by laoraryn** [reply]
Sounds good! I'll try shooting them an email to see what they said. Thank you so much for the help!
